demo/src/lint.test.ts

```diff
@@ -1,39 +1,39 @@
 import * as lint from '../../build'
 import * as path from 'path'
 import * as fs from 'fs'
-
-function log(relativeFilepath: string, diagnostics: lint.Diagnostic[]) {
+ 
+function log(relativeFilepath: string, diagnostics: readonly lint.Diagnostic[]) {
 	if (diagnostics.length <= 0) {
 		console.log('[No errors]\n')
 
 		return
 	}
 
 	for (const diag of diagnostics)
-		console.log(`${relativeFilepath}${diag.location ? `:${diag.location.first_line}:${diag.location.first_column}` : ''}: ${diag.message}`)
+		console.log(`${relativeFilepath}${diag.location ? `:${diag.location.coords?.first_line}:${diag.location.coords?.first_column}` : ''}: ${diag.message}`)
 
 	console.log('')
 }
 
 async function main() {
 	const filename = path.join(__dirname, '../resources/view.html')
 	const relFilename = path.relative(process.cwd(), filename).replace(/\\/g, '/')
-
 	const textDoc = fs.readFileSync(filename).toString()
+	if (!textDoc)
+		throw new Error(`Cannot open file '${filename}'`)
```

Comment:
`fs` throws exceptions, not empty strings. Use try statement.

Reply:
Removed